python/lsst/meas/extensions/piff/piffPsfDeterminer.py

```diff
@@ -0,0 +1,275 @@
+# This file is part of meas_extensions_piff.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__all__ = ["PiffPsfDeterminerConfig", "PiffPsfDeterminerTask"]
+
+import logging
+
+import numpy as np
+import piff
+import galsim
+
+import lsst.log
+import lsst.pex.config as pexConfig
+import lsst.meas.algorithms as measAlg
+from lsst.meas.algorithms.psfDeterminer import BasePsfDeterminerTask
+from .piffPsf import PiffPsf
+import lsst.afw.image as afwImage
+import lsst.afw.math as afwMath
+
+
+class PiffPsfDeterminerConfig(BasePsfDeterminerTask.ConfigClass):
+    spatialOrder = pexConfig.Field(
+        doc="specify spatial order for PSF kernel creation",
+        dtype=int,
+        default=2,
+    )
+    sizeCellX = pexConfig.Field(
+        doc="size of cell used to determine PSF (pixels, column direction)",
+        dtype=int,
+        default=256,
+        check=lambda x: x >= 10,
+    )
+    sizeCellY = pexConfig.Field(
+        doc="size of cell used to determine PSF (pixels, row direction)",
+        dtype=int,
+        default=sizeCellX.default,
+        check=lambda x: x >= 10,
+    )
+    samplingSize = pexConfig.Field(
+        doc="Resolution of the internal PSF model relative to the pixel size; "
+        "e.g. 0.5 is equal to 2x oversampling",
+        dtype=float,
+        default=1,
+    )
+    outlierNSigma = pexConfig.Field(
+        doc="n sigma for chisq outlier rejection",
+        dtype=float,
+        default=4.0
+    )
+    outlierMaxRemove = pexConfig.Field(
+        doc="Max fraction of stars to remove as outliers each iteration",
+        dtype=float,
+        default=0.05
+    )
+    maxSNR = pexConfig.Field(
+        doc="Rescale the weight of bright stars such that their SNR is less "
+            "than this value.",
+        dtype=float,
+        default=200.0
+    )
+
+    def setDefaults(self):
+        self.kernelSize = 21
+        self.kernelSizeMin = 11
+        self.kernelSizeMax = 35
+
+
+def computeWeight(maskedImage, maxSNR):
```

Comment:
Is the math behind this (and ,maybe more importantly, `applyMaxSNR`) documented anywhere already?

Reply:
It's documented within Piff, but probably worth copying over here.  Will do.